### ISSUE TITLE
Custom Attributes set in connected app aren't set in the Mobile SDK

### ISSUE THREAD

**Issue by timcolla** [forcedotcom/SalesforceMobileSDK-iOS#2958]
Hi,
I'm trying to set some custom attributes in the connected app and getting them through the MobileSDK, but it doesn't seem to work.
Thanks,
Tim

1. Version of Mobile SDK Used: 7.1.0
2. Issue found in Native App or Hybrid App: Native App
3. OS Version: iOS 12.4
4. Device: iPhone XR
5. Steps to reproduce:
- Use `AuthHelper.loginIfRequired { }` to bring up the login screen
- Log in through the presented WebView
- `UserAccountManager.shared.currentUserAccount?.customDataObject(forKey: "keyName")`
6. Actual behavior:
- customData is nil (keyName can't be found, and setting a breakpoint in the Mobile SDK shows that self->_customData is nil)
7. Expected Behavior:
- My "keyName" set in the connected app custom attributes is set
8. Error Log:
N/A

**Comment by trooper2013** [comment]
@timcolla The Custom attributes for your connected app is not accessible via the SFUserAccount. You need to set the keys that you are looking for in the ` [SFUserAccountManager sharedInstance].additionalOAuthParameterKeys`. These will be available in the dictionary  `additionalOAuthFields` which is accessible through the `userAccount.credentials`.

**Comment by timcolla** [comment]
Thanks for the quick reply @trooper2013 

I can't get this to work however. I've set the `additionalOAuthParameterKeys` and I've stepped through the SDK with breakpoints to make sure it's properly set.

The problem now is that the response in `handleUserAgentResponse:` doesn't contain the custom attribute set in the connected app.

I've set up the sdk this way:
`UserAccountManager.shared.additionalOAuthParameterKeys = ["customKey"]
            SalesforceManager.initializeSDK()`

And then I do this:
`            AuthHelper.loginIfRequired {
                print("Custom data")
                print(UserAccountManager.shared.currentUserAccount?.credentials.additionalOAuthFields)
            }`

But this is empty. I'm guessing because the custom attribute is not present in the response. 

Any idea what I might be doing wrong here?

Many thanks,
Tim
